packages/gitbook/src/components/DocumentView/Expandable/Details.tsx

```diff
@@ -0,0 +1,78 @@
+'use client'
+
+import { useParams } from 'next/navigation';
+import React from 'react';
+
+import { ClassValue, tcls } from '@/lib/tailwind';
+
+
+/**
+ * Details component rendered on client so it can expand dependent on url hash changes.
+ */
+export function Details(props: { children : React.ReactNode; id: string; contentIds?: string[]; open?: boolean; className?: ClassValue; }) {
+    const { children, id, contentIds = [], open, className } = props;
+    const params = useParams();
+    const [hash, setHash] = React.useState<string | null>(null);
+    
+    React.useEffect(() => {
+        function updateHash() {
+            const urlHash = global?.location?.hash;
+            if (urlHash !== hash) {
+                setHash(urlHash.slice(1));
+                const element = document.getElementById(urlHash.slice(1));
+                if (element) {
+                    element.scrollIntoView({
+                        block: 'start',
+                        behavior: 'smooth',
+                    });
+                }
+            }
+        }
+        addEventListener('hashchange', updateHash);
+        updateHash();
+        return () => removeEventListener('hashchange', updateHash); 
+    }, [params, hash, open])
```

Comment:
Is there no hook from NPM or Next we can use here?

It looks like we'll need this in a bunch of places

Reply:
I've separated out the useHash hook. I couldn't find anything that did this exactly, with the params dependency.